server/src/main/java/io/crate/protocols/http/Headers.java

```diff
@@ -64,15 +64,30 @@ public static void setKeepAlive(HttpVersion httpVersion, FullHttpResponse resp)
         }
     }
 
-    public static Credentials extractCredentialsFromHttpBasicAuthHeader(String authHeaderValue) {
+    /**
+     * An entry point for HTTP authentication, forwards to either Basic or JWT, depending on header.
+     * @param authHeaderValue contains authentication schema (basic or bearer) and auth payload (token or password) separated by space.
+     * Authentication schema is case-insensitive: <a href="https://datatracker.ietf.org/doc/html/rfc7235?ref=blog.teknkl.com#section-2.1">spec</a>
+     */
+    public static Credentials extractCredentialsFromHttpAuthHeader(String authHeaderValue) {
         if (authHeaderValue == null || authHeaderValue.isEmpty()) {
             // Empty credentials.
-            return new Credentials("", new SecureString(new char[] {}));
+            return new Credentials(null, null);
         }
+        String[] splitHeader = authHeaderValue.split(" ");
+        assert splitHeader.length == 2 :
+            "Header must contain only authentication schema and base64 encoded value, separated by a whitespace";
```

Comment:
```suggestion
            "Header must contain only authentication scheme and base64 encoded value, separated by a whitespace";
```

Reply:
in docs with name it  `method`, so maybe change it to `method` here as well?

Reply:
`method` in HBA is specific for our auth methods.
scheme is headers "key" name, either basic -> password or bearer -> jwt.
it's used in spec, https://datatracker.ietf.org/doc/html/rfc7235?ref=blog.teknkl.com#section-2.1. I adjusted header javadocs.